x/group/server/invariants.go

```diff
@@ -0,0 +1,98 @@
+package server
+
+import (
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/regen-network/regen-ledger/orm"
+	"github.com/regen-network/regen-ledger/x/group"
+)
+
+func (s serverImpl) RegisterInvariants(ir sdk.InvariantRegistry) {
+	ir.RegisterRoute(group.ModuleName, "Tally-Votes", s.tallyVotesInvariant())
+}
+
+func (s serverImpl) AllInvariants() sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		return s.tallyVotesInvariant()(ctx)
+	}
+}
+
+func (s serverImpl) tallyVotesInvariant() sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var msg string
+		var broken bool
+		if ctx.BlockHeight()-1 < 0 {
+			return sdk.FormatInvariant(group.ModuleName, "Tally-Votes", "Not enough blocks to perform TallyVotesInvariant"), false
+		}
+		prevCtx := ctx.WithBlockHeight(ctx.BlockHeight() - 1)
+		prevIt, err := s.proposalTable.PrefixScan(prevCtx, 1, math.MaxUint64)
+		if err != nil {
+			panic(err)
+		}
+		curIt, err := s.proposalTable.PrefixScan(ctx, 1, math.MaxUint64)
+		if err != nil {
+			panic(err)
+		}
+
+		var t require.TestingT
+		var curProposals []*group.Proposal
+		_, err = orm.ReadAll(curIt, &curProposals)
+		require.NoError(t, err, &curProposals)
```

Comment:
why do we have this test `require` in the middle of some regular code?

Reply:
I'll remove that